### ISSUE TITLE
[sonos] From echo, notification with say-command not always played! 

### ISSUE THREAD

**Issue by JueBag** [openhab/openhab-addons#6818]
Playing a TTS notification sound isn't working under some cirumstances, probably cause by a changed behaviour on how the echo device selects and sends the URI to the SONOS devices.

For german public radio stations ("öffentlich-rechtliche") Amazon is using now the "ARD Audio Mediathek" instead of TuneIn, When asking for such a station the Echo device installs the respective skill to use this source automatically. IMHO this change is causing the bug.

I did some tests and logged the corresponding current Track URI and current AV Transport URI:
Running openHAB 2.5 
No change on "other" radio stations:
TuneIn Radiostation [(Absolute relax (Easy listening)]
Started via Sonos App:

AV Transport URI:
`x-sonosapi-stream:s158432?sid=254&flags=8224&sn=0`

Track URI:
`x-rincon-mp3radio://http://absolut.hoerradar.de/absolutradio-relax-mp3?sABC=5r1nro98%230%23sn0889oq88q9op892o6qrorn02149r11%23gharva&amsparams=playerid:tunein;skey:1578822552`
Say Command working!

Started via Echo:

AV Transport URI:
`x-sonosapi-stream:s56851?sid=254&flags=8224&sn=0`

Track URI:
`x-rincon-mp3radio://http://absolut.hoerradar.de/absolutradio-relax-mp3?sABC=5r1ns790%230%23sn0889oq88q9op892o6qrorn02149r11%23gharva&amsparams=playerid:tunein;skey:1578825616`
Say Command working!

German Radio Station
Radiostation NDR2 

Started via Sonos App using TuneIn

AV Transport URI:
`x-sonosapi-stream:s56851?sid=254&flags=8224&sn=0`

Track URI:
`x-rincon-mp3radio://http://ndr-edge-2040.fra-lg.cdn.addradio.net/ndr/ndr2/niedersachsen/mp3/128/stream.mp3?ar-distributor=f0a1`
Say Command working!

Started via Sonos App using a link provided by radiostation, although SonosApp still states TuneIn Radio!

AV Transport URI:
`x-rincon-mp3radio://http://www.ndr.de/resources/metadaten/audio/m3u/ndr2.m3u`

Track URI:

` x-rincon-mp3radio://http://ndr-edge-2065.fra-lg.cdn.addradio.net/ndr/ndr2/niedersachsen/mp3/128/stream.mp3`
Say Command **NOT** working! This is more a "secondary" problem!

Started via Echo: (Echo states its from the „ARD Audio Mediathek“, Sonos shows ARD instead of TuneIn)

AV Transport URI:
`x-rincon-mp3radio://https://ndr-ndr2-hamburg.sslcast.addradio.de/ndr/ndr2/hamburg/mp3/128/stream.mp3?ar-distributor=f0e2`

Track URI:
`x-rincon-mp3radio://https://ndr-edge-10b2-fra-dtag-cdn.cast.addradio.de/ndr/ndr2/hamburg/mp3/128/stream.mp3?ar-distributor=f0e2`
Say Command **NOT** working!

**Comment by lolodomo** [comment]
Ok I see in the code why notification is not working in this case. Only 
x-rincon-mp3radio with particular URL (tunein I guess) is supported. 
I will try to fix that.

**Comment by JueBag** [comment]
Thanks, but don't hurry!

**Comment by lsiepel** [comment]
@lolodomo did you ever get the change to fix it, or is it still an issue?

**Comment by lolodomo** [comment]
No, this was probably either forgotten or I did not find a proper solution, I don't remember.

**Comment by lsiepel** [comment]
Might this something for you to look at @jlaur as you are doing some work on the sonos binding?

**Comment by lsiepel** [comment]
Super old issue, probably no longer relevant as there have not been any newer reports.

**Comment by lolodomo** [comment]
Old but may be still relevant.
